File: app/src/main/java/com/esafirm/androidplayground/common/MenuListAdapter.java
package com.esafirm.androidplayground.common;

import android.content.Context;
import android.widget.ArrayAdapter;

import java.util.List;

public class MenuListAdapter extends ArrayAdapter<String> {

    public MenuListAdapter(Context context) {
        super(context, android.R.layout.simple_list_item_1);
    }

    public void setData(List<String> data) {
        clear();
        addAll(data);
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/esafirm/androidplayground/common/MenuListAdapter.kt
package com.esafirm.androidplayground.common

import android.content.Context
import android.widget.ArrayAdapter

class MenuListAdapter(
    context: Context
) : ArrayAdapter<String>(context, android.R.layout.simple_list_item_1) {

    fun setData(data: List<String>) {
        clear()
        addAll(data)
        notifyDataSetChanged()
    }
}
